### ISSUE TITLE
CHAOS XGPSCN

### ISSUE THREAD

**Issue by larsbrinkhoff** [PDP-10/its#2098]
@BobKerns' SCAN program (#636) should also be linked from DEVICE; CHAOS XGPSCN.

**Comment by larsbrinkhoff** [comment]
As a client, it's hardwired to connect to AI and knows it's address.  That's not very useful now.

```
aichad==:2026                      ; AI chaosnet address
```

**Comment by BobKerns** [comment]
How would you like to discover where the XGP is connected?

**Comment by larsbrinkhoff** [comment]
A simple solution, common in ITS programs, would be to have it a constant which can be (re)defined during assembly.  A small step up from that might be to have the constant be a hostname that is looked up in a table or dynamically with DNS.

**Comment by BobKerns** [comment]
Except, there is no XGP, right? Is there an emulated one that it can talk to?

It depends on chaosnet; do I recall there's an encapsulation? IIRC it's just a stream, though. How would it interface to DNS?

I can't see a table as being helpful.

It seems kind of pointless to make it configurable, if there was only ever one XGP, attached to AI, and there would never be another. If there's an emulation available now, after nearly 50 years, that assumption would no longer hold!

The networking is only useful if you have two ITS's, one with XGP, one without. Otherwise, we could just hardware it to always be local. Which would nullify the one cool thing about it, but...

**Comment by larsbrinkhoff** [comment]
There isn't an emulated XGP yet, but it's one of the top items on my must-have list.  At some point in the future there should be an ITS online with an XGP.  Yes, there is encapsulations for Chaosnet.  There are several, but the one most useful for talking across the internet is through UDP.

For sure, there is only a tiny shred of an actual use case for adding the XGPSCN Chaosnet service, but I like to make all software available whether it's reasonable or not.

**Comment by eswenson1** [comment]
What mechanism do we have on ITS now to print? I think it would be great to have an emulated XGP so that we can print arbitrary files, TeX output, @ listings, Macsyma graphs, etc. Can we do that now, somehow? If not, then XGP would be really nice. 

For Multics, we can generate PDF files from (text) lineprinter spooled segments (files) using a prt2pdf program running on the host. It would be nice to have an XGP2PDF utility.

**Comment by larsbrinkhoff** [comment]
Yes, that would be great.  For now there's only @atsampson's "harscntopdf" script as described here: https://github.com/PDP-10/its/blob/master/doc/printing.md

**Comment by larsbrinkhoff** [comment]
> Except, there is no XGP, right? Is there an emulated one that it can talk to?

There is now!
